apps/src/templates/HourOfCodeGuideEmailDialog.jsx

```diff
@@ -99,11 +143,9 @@ function HourOfCodeGuideEmailDialog({isSignedIn}) {
             />
             <Button
               id="uitest-email-guide"
-              text={emailGuideButtonText}
+              text={isSendInProgress ? i18n.inProgress() : emailGuideButtonText}
               onClick={() => {
-                saveInputs();
-                sendEmail();
-                onClose();
+                validateAndSave();
```

Comment:
I like the way this was all restructured! I think this line in particular could also be simplified to something like `onClick={validateAndSave}`

Reply:
good catch! I updated this one and the onClose above.